docs/v2_migration.md

```diff
@@ -55,6 +55,17 @@ When different build tools are use, at least one layout needs to be set.
 The `src_folder` must be the same when using different layouts and should
 not depend on settings or options.
 
+## CMakeToolchain
+
+The old `CMake.definition` should be replaced by `CMakeToolchain.variables` and move to the `generate` method.
+However, there are two main cases to use `cache_variables`:
+
+- When a definition is configured before project()
+- When an option uses CACHE. Need to check project's CMakeLists.txt
```

Comment:
There's two CMake commands used to set cache variables in the majority of cases:

```cmake
option(...)
set(... CACHE...)
```

It's probably worth mentioning those specifically.
Additionally, a lot of built-in CMake modules and functions will end up setting cache variables as well. The `find_*()` functions end up setting cache variables. Including the `CTest` module automatically adds the `BUILD_TESTING` cache variable as well.

It might be easier to run the CMake configure command and check if a variable shows up in `CMakeCache.txt` in the build directory. If it does, it's a cache variable.